.github/workflows/agw-build-publish-container.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: AGW Build & Publish Container
+
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+  pull_request_target:
+    types: [ opened, reopened, synchronize ]
+    branches:
+      - master
+      - 'v1.*'
+
+jobs:
+
+  collect-is-successful:
+    runs-on: ubuntu-latest
+    if: success()
+    needs: build-containers
+    steps:
+      - run: echo "Successfully built AGW containers."
+
+  check-is-successful:
+    runs-on: ubuntu-latest
+    if: always()
+    needs: collect-is-successful
+    steps:
+      - name: Check completion of all build steps
+        run: |
+          if [ "${{ needs.collect-is-successful.result }}" = "success" ];
+          then
+            exit 0
+          else
+            echo "Exit status of some previous job(s) was ${{ needs.collect-is-successful.result }}"
+            exit 1
+          fi
+
+  build-containers:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_REGISTRY: agw-ci.artifactory.magmacore.org
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - DOCKER_IMAGE: gateway_go
+            DOCKER_FILE: feg/gateway/docker/go/Dockerfile
+          - DOCKER_IMAGE: agw_gateway_python
+            DOCKER_FILE: lte/gateway/docker/services/python/Dockerfile
+          - DOCKER_IMAGE: agw_gateway_c
+            DOCKER_FILE: lte/gateway/docker/services/c/Dockerfile
+
+    steps:
+      - run: echo "DOCKER_REGISTRY=agw-test.artifactory.magmacore.org" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master'
```

Comment:
Do we not need to check here that the repository owner is `magma`? Won't this fail if someone opens a PR from their fork master branch?

Reply:
No. No, there is a safeguard in the composite action for that.

Reply:
https://github.com/magma/magma/blob/dd0017c07c7c2d14ea5a49dbcd9439278136ee64/.github/workflows/composite/docker-builder-agw/action.yml#L61